vcf/src/parse.rs

```diff
@@ -15,11 +17,22 @@ impl<'src> Header<'src> {
 
 impl<'src> HeaderValue<'src> {
     pub fn parse(input: &'src str) -> Result<Self, ParseError> {
+        // Repeatedly match either non-comma/non-quote characters or blocks of text enclosed in
+        // quotes, until we can't, in which case we're either at a non-quote-enclosed comma or the
+        // end of the string.
+        let re = Regex::new(r#"(?:[^,"]+|(?:"[^"]*"))+"#).unwrap();
```

Comment:
This regex gets compiled every time the function gets called, which doesn't seem very efficient. How do I place it outside the function call so that it is only compiled once?

Reply:
Use something like lazy_static!:

https://stackoverflow.com/a/35169402

Or https://github.com/rust-lang/regex/issues/709#issuecomment-1456381501 which uses std rust. Seems like there's something in the works to compile statically actually in the regex crate which would be nice.

Reply:
Using `lazy_static!` seems to have done the job. Cheers.